hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azure/ITestWasbRemoteCallHelper.java

```diff
@@ -434,10 +431,6 @@ class MatchesPattern extends TypeSafeMatcher<String> {
       }
     }
 
-    expectedEx.expectMessage(new MatchesPattern(
```

Comment:
Did we lose an expectation here?

Reply:
In JUnit 5, `org.junit.rules.ExpectedException` has been deprecated, and it is recommended to use assertThrows to validate exception throwing, as shown below:

```
import static org.junit.jupiter.api.Assertions.assertThrows;

@Test
public void shouldThrowException() {
  assertThrows(IllegalArgumentException.class, () -> {
      someMethodThatThrows();
  });
}
```

The change is primarily in syntax, where assertThrows captures exceptions via lambda expressions, replacing the ExpectedException rule from JUnit 4. However, it does not have any fundamental impact on the logic of unit tests.

I can further improve this unit test to also validate the exception message using a regular expression.

Reply:
I have seen the conversions to `assertThrows`. Those all look good.

Perhaps what I find confusing is that the `setupExpectations()` method now seems to be a no-op. It seems to define a custom matcher class, but then never use it. Should `setupExpectations()` be removed, or am I missing something?

Reply:
Thank you for your suggestion! I also believe that `setupExpectations` should be removed. I will continue to improve the code.